src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -150,5 +155,96 @@ public void ConnectHeaders_NewAndExistingHeaders_ShouldCreateNewHeaders()
             Assert.Equal(listOfNewHeaders.Last(), connectedHeadersNew.DownloadTo.Header);
             Assert.Equal(listOfNewHeaders.First(), connectedHeadersNew.DownloadFrom.Header);
         }
+
+        // Supply headers that we already have, make sure no new ChainedHeaders were created.
+        [Fact]
+        public void ConnectHeaders_SupplyExistingHeaders_DontCreateNewChainHeaders()
+        {
+            var testContext = new TestContext();
+            ChainedHeaderTree cht = testContext.CreateChainedHeaderTree();
+            ChainedHeader chainTip = testContext.ExtendAChain(5);
+            Assert.Empty(cht.GetChainedHeadersByHash());
+
+            cht.Initialize(chainTip, true);
+
+            Assert.Equal(6, cht.GetChainedHeadersByHash().Keys.Count);
+
+            var beforeKeys = new List<uint256>(cht.GetChainedHeadersByHash().Keys);
+
+            List<BlockHeader> listOfExistingHeaders = testContext.ChainedHeaderToList(chainTip, 5);
+            cht.ConnectNewHeaders(1, listOfExistingHeaders);
```

Comment:
Why would you define those constants for peer identifiers if you don't use them then?